File: ExampleMvvm/librarycommon/src/main/java/com/degde/library/library/utils/Utilerias.java
package com.degde.library.library.utils;

import android.app.Activity;
import android.app.AlertDialog;
import android.content.Context;
import android.content.DialogInterface;
import android.content.Intent;

public class Utilerias {
    public void openActivity(Context context, Class cl){
        Intent intent = new Intent(context, cl);
        context.startActivity(intent);
        ((Activity)context).finish();
    }
    public void showMessageError(Context context){
        AlertDialog.Builder builder1 = new AlertDialog.Builder(context);
        builder1.setTitle("Alerta");
        builder1.setMessage("Problemas de conectividad");
        builder1.setCancelable(true);

        builder1.setPositiveButton("Entendido", new DialogInterface.OnClickListener() {
            public void onClick(DialogInterface dialog, int id) {
                dialog.cancel();
            }
        });
        AlertDialog alert = builder1.create();
        alert.show();
    }
}

File: ExampleMvvm/corekotlin/src/main/java/com/core/mvvm/utils/Utilerias.kt
package com.core.mvvm.utils

import android.app.Activity
import android.app.AlertDialog
import android.content.Context
import android.content.Intent


class Utilerias {
    fun openActivity(context: Context, cl: Class<*>?) {
        val intent = Intent(context, cl)
        context.startActivity(intent)
        (context as Activity).finish()
    }

    fun showMessageError(context: Context?) {
        val builder1 = AlertDialog.Builder(context)
        builder1.setTitle("Alerta")
        builder1.setMessage("Problemas de conectividad")
        builder1.setCancelable(true)
        builder1.setPositiveButton("Entendido") { dialog, id -> dialog.cancel() }
        val alert = builder1.create()
        alert.show()
    }
}
